types/node/worker_threads.d.ts

```diff
@@ -701,7 +701,7 @@ declare module "worker_threads" {
      * @param value Any arbitrary, cloneable JavaScript value that will be cloned and passed automatically to all new `Worker` instances. If `value` is passed as `undefined`, any previously set value
      * for the `key` will be deleted.
      */
-    function setEnvironmentData(key: Serializable, value: Serializable): void;
+    function setEnvironmentData(key: Serializable, value: Serializable | undefined): void;
```

Comment:
The API actually just defines this parameter as optional.


```suggestion
    function setEnvironmentData(key: Serializable, value?: Serializable): void;
```

Reply:
Thanks for the quick reply @Renegade334!

To me it feels more sensible to accept undefined rather than be optional - e.g. we're intentionally removing the key. Making it optional could lead to accidentally deleting a key in the case somebody forgot to pass a value to set.

With that in mind, would you like me to make all declarations of this function use the non-optional behaviour?

Reply:
@types/node doesn't aim to constrain usage that the Node API itself considers valid, although I appreciate the argument. If it's documented as optional upstream, it should really be optional here.

Reply:
For context, this is specifically a tested case:

https://github.com/nodejs/node/blob/283ed5357fee095a05339775a9f75b7c7a7e92fc/test/parallel/test-worker-environmentdata.js#L25

Reply:
Sounds good. I've made them optional and updated the v18 and v20 trees.